packages/typescript/server/src/main.ts

```diff
@@ -59,28 +57,89 @@ export class Jamsocket {
   }
 
   async connect(connectRequest?: ConnectRequest): Promise<ConnectResponse> {
-    const response = await fetch(
-      `${this.apiUrl}/v2/service/${this.account}/${this.service}/connect`,
-      {
+    const response = checkResponse(
+      'connect()',
+      await fetch(`${this.apiUrl}/v2/service/${this.account}/${this.service}/connect`, {
         method: 'POST',
         headers: { Authorization: `Bearer ${this.token}`, 'Content-Type': 'application/json' },
         body: JSON.stringify(connectRequest || {}),
         cache: 'no-store',
-      },
+      }),
     )
-    const bodyText = await response.text()
-    if (!response.ok) {
-      if (response.status === 429) {
-        console.warn(
-          "You've hit the spawn rate limit. This may be because you've spawned too many session backends in a short period of time or you're already running the maximum number of concurrent session backends. (related: https://docs.jamsocket.com/pricing/free-tier-limits)",
-        )
-      }
-      throw new Error(`Error spawning backend: ${response.status} ${bodyText}`)
+    return parseAs<ConnectResponse>(isConnectResponse, await response.text())
+  }
+
+  async status(backendId: string): Promise<BackendState> {
+    const url = `${this.apiUrl}/v2/backend/${backendId}/status`
+    const response = checkResponse('status()', await fetch(url, { cache: 'no-store' }))
+    return parseAs<BackendState>(isBackendState, await response.text())
+  }
+
+  async statusStream(backendId: string, onStatus: OnStatusCallback): Promise<UnsubscribeFn> {
+    const url = `${this.apiUrl}/v2/backend/${backendId}/status/stream`
+
+    // keep track of the statuses we've seen since we might have just resubscribed
+    // to the status stream and are seeing some of the statuses we've already seen again
+    const statusesSeen = new Set<BackendStatus>()
+    let unsubscribed = false
+    let retries = 0
+    let streamReader: ReadableStreamDefaultReader<string> | null = null
+    function unsubscribe() {
+      unsubscribed = true
+      streamReader?.cancel()
+      streamReader = null
     }
-    try {
-      return JSON.parse(bodyText) as ConnectResponse
-    } catch (e) {
-      throw new Error(`Error parsing connect response: ${bodyText}`)
+
+    async function startStatusStream() {
+      try {
+        const response = checkResponse('statusStream()', await fetch(url, { cache: 'no-store' }))
```

Comment:
IIRC from the y-sweet SDK, `{cache: 'no-store'}` makes this non-portable outside of next.js. The workaround we found was to add a random number as an unused query string.

Could probably also be omitted in this case because I don't think streams are cached.

Reply:
Ah yeah, I think it's just Cloudflare Workers that this throws in. I'll update

Reply:
Also, it looks like this issue was fixed in a more recent version of NextJS (`cache` is `no-store` by default now). So I think I'll drop the cache-busting techniques across the board.